packages/botonic-plugin-nlu/src/nlu.js

```diff
@@ -1,139 +1,58 @@
-import * as tf from '@tensorflow/tfjs'
 import { default as fetch } from 'node-fetch'
-import path from 'path'
-import { readFile, readJSON } from './fileUtils'
-import { Tokenizer, padSequences, detectLang } from './preprocessing'
-import { processEntities } from './ner'
+import { resolveEnv, loadOption } from './utils'
+import { detectLang } from '@botonic/nlu/lib/preprocessing'
 import {
-  NLU_DATA_FILENAME,
-  MODEL_FILENAME,
-  MODELS_DIRNAME,
-  NLU_DIRNAME,
-  ASSETS_DIRNAME
-} from './constants'
-
-const isBrowser = () => {
-  return typeof window !== 'undefined' && !window.process
-}
-
-const isProd = () => {
-  return process.env.STATIC_URL !== undefined
-}
+  getIntent,
+  getEntities,
+  getPrediction
+} from '@botonic/nlu/lib/prediction'
 
 global.fetch = fetch
 
-async function resolveEnv() {
-  if (isProd() || isBrowser()) {
-    if (isProd()) {
-      return {
-        mode: 'prod',
-        uri: `${process.env.STATIC_URL}/${ASSETS_DIRNAME}/${MODELS_DIRNAME}/`
-      }
-    } else {
-      return { mode: 'dev', uri: window.location.href }
-    }
-  } else {
-    return {
-      mode: 'node',
-      uri: `${path.join(process.cwd(), 'src', NLU_DIRNAME, MODELS_DIRNAME)}/`
-    }
-  }
-}
-
-function loadOption(lang, env) {
-  let nlu = {}
-  try {
-    if (env.mode === 'prod' || env.mode === 'dev') {
-      nlu.nluData = axios({
-        url: `${env.uri}${lang}/${NLU_DATA_FILENAME}`
-      })
-      nlu.model = tf.loadLayersModel(`${env.uri}${lang}/${MODEL_FILENAME}`)
-    } else {
-      nlu.nluData = readJSON(`${env.uri}/${lang}/${NLU_DATA_FILENAME}`)
-      nlu.model = tf.loadLayersModel(
-        `file://${env.uri}${lang}/${MODEL_FILENAME}`
-      )
-    }
-  } catch (e) {
-    console.log('Cannot retrieve NLU Information', e)
-  }
-  return nlu
-}
-
 export class NLU {
-  constructor(options) {
+  constructor(languages) {
     return (async () => {
       this.env = await resolveEnv()
       this.langs = []
-      this.nlus = {}
-      for (let config of options) {
-        this.langs.push(config.LANG)
-        let { nluData, model } = loadOption(config.LANG, this.env)
-        this.nlus[config.LANG] = {
-          lang: config.LANG,
+      this.models = {}
+      for (let language of languages) {
+        this.langs.push(language)
+        let { nluData, model } = loadOption(language, this.env)
+        this.models[language] = {
+          lang: language,
           model: model,
           nluData: nluData
         }
       }
       await Promise.all([
-        ...Object.values(this.nlus).map(nlu => nlu.model),
-        ...Object.values(this.nlus).map(nlu => nlu.nluData)
+        ...Object.values(this.models).map(nlu => nlu.model),
+        ...Object.values(this.models).map(nlu => nlu.nluData)
       ])
-      for (let [k, v] of Object.entries(this.nlus)) {
-        let nluData = await v.nluData
-        let { intentsDict, maxSeqLength, vocabulary } =
+      for (let [lang, data] of Object.entries(this.models)) {
```

Comment:
in the line above you called it nlu instead of data, right?

Reply:
this data refers to the values which are resolved from the previous promises, i have renamed it to res to avoid confusions